controllers/containerimagebuild_controller.go

```diff
@@ -149,56 +141,65 @@ func (r *ContainerImageBuildReconciler) SetupWithManager(mgr ctrl.Manager) error
 		Complete(r)
 }
 
-func (r *ContainerImageBuildReconciler) getAuthCredentials(ctx context.Context, log logr.Logger, registry forgev1alpha1.Registry) (username, password string, err error) {
-	switch registry.BasicAuth {
-	case "":
-		log.Info("Not using registry credentials")
-		return
-	case forgev1alpha1.BasicAuthInline:
-		log.Info("Processing inline registry credentials")
-		username = registry.Username
-		password = registry.Password
-	case forgev1alpha1.BasicAuthSecret:
-		log.Info("Processing secret registry credentials")
-		var secret corev1.Secret
-		if err = r.Client.Get(ctx, types.NamespacedName{
-			Namespace: registry.SecretNamespace,
-			Name:      registry.SecretName,
-		}, &secret); err != nil {
-			err = fmt.Errorf("cannot find registry auth secret: %v", err)
-			return
-		}
+// NOTE: the following 2 funcs should not really be attached to the Reconciler
 
-		actual, expected := secret.Type, corev1.SecretTypeDockerConfigJson
-		if actual != expected {
-			err = fmt.Errorf("registry auth secret type must be %v, not %v", expected, actual)
-			return
+func (r *ContainerImageBuildReconciler) buildRegistryConfig(ctx context.Context, apiRegs []forgev1alpha1.Registry) ([]config.Registry, error) {
+	var configs []config.Registry
+	for _, apiReg := range apiRegs {
+		regConf := config.Registry{
+			Host:   apiReg.Server,
+			NonSSL: apiReg.NonSSL,
 		}
 
-		input := secret.Data[corev1.DockerConfigJsonKey]
-		var output credentials.DockerConfigJSON
-		if err = json.Unmarshal(input, &output); err != nil {
-			err = fmt.Errorf("cannot parse docker config in registry secret: %v", err)
-			return
-		}
-
-		auth, ok := output.Auths[registry.URL]
-		if !ok {
-			var urls []string
-			for k, _ := range output.Auths {
-				urls = append(urls, k)
+		// NOTE: should we validate?
```

Comment:
Why not?

Reply:
I think we should validate but I guess I didn't outline my thoughts here. We can leverage and [admission webhook](https://book.kubebuilder.io/reference/admission-webhook.html) to validate objects before they get registered and I think that's the cleanest way to front the validation of input.

There's a little more setup that's required for that but I don't have the time to pay that price right now, so I'll just validate here for now and add a note about that in the code.